### DISCUSSION TITLE
What does your E2E testing setup look like?

### DISCUSSION THREAD

**Discussion by canrozanes** [karlkeefer/pngr#42] [Q&A]
Shipping features without tests give me anxiety. 

I'm familiar with using a library like [Mirage.js](https://miragejs.com/) to mock the back-end and write proper UI tests.  However, PNGR is a full-stack app and I do have access to the back-end locally. So I was thinking, a setup where I seed the actual backend with real data and use that data to write assertions on the UI would be a lot better than testing on a fake back-end.

Does anyone have such a setup? Having done some googling, looks like I could use a tool like [Cypress](https://docs.cypress.io/guides/overview/why-cypress) to do something like that but before going down in a rabbit hole with Cypress, I figured I'd ask the community.

What does your E2E testing setup look like?

**Comment by karlkeefer** [comment]
I've not setup e2e on any of my PNGR projects.
If you find a clean way to do this, I would definitely be down to merge something in as a template for all future forks.

I'd be inclined to try to get something working with [playwright](https://github.com/microsoft/playwright) over Cypress, just because I think it'd work better in our docker + ci use-cases, but I haven't used either so this is only based on what others have written comparing them.

**Reply by karlkeefer** [reply]
You could hook into the github actions to run e2e tests in CI: https://github.com/karlkeefer/pngr/blob/master/.github/workflows/build.yml
You might also need to make a `docker-compose.test.yml` to trigger something to run the e2e tests with all the other containers _up_.

I personally feel like the tests runs have to be hooked into the build server... running them in dev just isn't enough.
